sal-common/src/main/java/org/ow2/proactive/sal/model/PACloud.java

```diff
@@ -102,13 +105,34 @@ public class PACloud implements Serializable {
     @ElementCollection(targetClass = String.class)
     private Map<String, String> deployedWhiteListedRegions;
 
-    @JsonManagedReference(value = "pacloudReference")
     @OneToMany(fetch = FetchType.EAGER, orphanRemoval = true, cascade = CascadeType.ALL)
+    @JsonIdentityReference(alwaysAsId = true)
+    @JsonProperty("deploymentNodeNames")
     private List<Deployment> deployments;
 
     @OneToOne
     private Credentials credentials;
 
+    public static PACloud fromId(String cloudId) {
+        PACloud paCloud = null;
+        ApplicationContext applicationContext = SpringConfiguration.contextProvider().getApplicationContext();
+        if (applicationContext != null) {
+            PACloudRepository paCloudRepository = (PACloudRepository) applicationContext.getBean("paCloudRepository");
+            paCloud = paCloudRepository.findOne(cloudId);
+        }
+        if (paCloud == null) {
+            paCloud = new PACloud();
+            paCloud.cloudId = cloudId;
+        }
+        return paCloud;
+    }
+
+    @JsonSetter("deploymentNodeNames")
+    public void setDeploymentsByIds(List<String> deployments) {
+        this.deployments = deployments.stream().map(Deployment::fromId).collect(Collectors.toList());
+        this.deployments.forEach(deployment -> deployment.setPaCloud(this));
+    }
```

Comment:
Same comment as upper. Also it would be better to create a method to retrieve deployments in bulk instead of a query for each